netbox/core/views.py

```diff
@@ -658,7 +658,7 @@ def test_func(self):
     def get(self, request):
         q = request.GET.get('q', None)
 
-        plugins = get_plugins().values()
+        plugins = get_plugins(request).values()
```

Comment:
We probably want to avoid passing the request through to a utility function solely for the purpose of collecting a potential error message. (We should preserve the option to fetch catalog plugins absent a WSGI request.)

Instead, I'd suggest refactoring the functions:

- Move the caching logic from `get_plugins()` into `get_catalog_plugins()`
- `get_catalog_plugins()` should return an empty dict if fetching fails
- Extend `get_local_plugins()` to accept an optional dictionary of plugins, to which local plugins are added/updated
- Check in the view for a failure to retrieve catalog plugins

Something like this:

```python
plugins = get_catalog_plugins()
if not plugins:
    messages.warning(request, _("Plugins catalog could not be loaded"))
plugins = get_local_plugins(plugins)
```